listen.js

```diff
@@ -10,24 +10,21 @@ const mockSSB = { backlinks: true, query: true }
 const { isUpdate: isPOBox } = new CRUT(mockSSB, poBoxSpec).spec
 
 module.exports = {
-  addMember (ssb) {
-    return pull(
-      ssb.messagesByType({ type: 'group/add-member', private: true, live: true }),
-      // NOTE this will run through all messages on each startup, which will help guarentee
-      // all messages have been emitted AND processed
-      // (same not true if we used a dummy flume-view)
-      pull.filter(m => m.sync !== true),
-      pull.filter(isAddMember),
-      pull.unique('key')
-      // NOTE we DO NOT filter our own messages out
-      // this is important for rebuilding indexes and keystore state if we have to restore our feed
-    )
-  },
-  excludeMember (ssb) {
+  mutateMembers (ssb) {
+    const query = [{
+      $filter: {
+        value: {
+          content: {
+            type: { $in: ['group/add-member', 'group/exclude-member'] }
+          }
+        }
+      }
+    }]
+
     return pull(
-      ssb.messagesByType({ type: 'group/exclude-member', private: true, live: true }),
+      ssb.query.read({ query, old: true, live: true }),
```

Comment:
:fire: this will LIKELY scan the whole database. With DB1 the way to check if you have a flume-view index supporting you, or if it's falling back to scan whole db is https://www.npmjs.com/package/ssb-query#queryexplain-query

```js
console.log(ssb.query.explain({ query, old: true, live: true }))
```


I think you should stick with `messagesByType` and use http://pull-stream.github.io/#pull-merge
NOTE: you **must** include a compare function with that otherwise you will eat shit. Just compare on receive time or something

Reply:
I think if you're trying to improve things by reducing the number of pull-streams that are live, then db2 will help with this. It's indexes are a lot more flexible

Reply:
i checked pull-merge but it said it removes duplicates, so doing it this way seemed easier

Reply:
i didn't want things to break if 2 msgs had the same timestamp